src/components/cc-email-list/cc-email-list.smart.js

```diff
@@ -0,0 +1,242 @@
+import {
+  todo_addEmailAddress as addEmailAddress,
+  todo_getEmailAddresses as getEmailAddresses,
+  todo_removeEmailAddress as removeEmailAddress,
+} from '@clevercloud/client/esm/api/v2/user.js';
+import { defineSmartComponent } from '../../lib/define-smart-component.js';
+import { i18n } from '../../lib/i18n.js';
+import { notify, notifyError, notifySuccess } from '../../lib/notifications.js';
+import { sendToApi } from '../../lib/send-to-api.js';
+import { CcEmailList } from './cc-email-list.js';
+import '../cc-smart-container/cc-smart-container.js';
+
+defineSmartComponent({
+  selector: 'cc-email-list',
+  params: {
+    apiConfig: { type: Object },
+  },
+  onContextUpdate ({ component, context, onEvent, updateComponent, signal }) {
+    updateComponent('emails', { state: 'loading' });
+    updateComponent('addEmailForm', CcEmailList.ADD_FORM_INIT_STATE);
+
+    const api = getApi(context.apiConfig, signal);
```

Comment:
Interesting... I'm not sure about this yet but I like the let's try to improve all this verbosity.

I'd like the opinion of the team on this and if positive maybe discuss doing this elsewhere.

Note that the API calls that modify stuffs should not use the signal as we still want them to go through even if there's a navigation. I'll add notes below.

Reply:
Some arguments for this:

* improve verbosity
* in the code, you manipulate a variable named `api` which guides the reader about what is involved
* If you want a temporary mocked `api` it may be easier to write.